helm/k8s-nginx-ingress/templates/nginx-ingress-controller.yaml

```diff
@@ -0,0 +1,66 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: "{{ .Values.service.name }}"
+  annotations:
+    service.beta.kubernetes.io/aws-load-balancer-ssl-cert: "{{ .Values.service.ssl.aws_certificate_arn }}"
+    service.beta.kubernetes.io/aws-load-balancer-ssl-ports: "443"
+    service.beta.kubernetes.io/aws-load-balancer-backend-protocol: http
+spec:
+  type: LoadBalancer
+  ports:
+    # Removed the plaintext port - but we could reinstate with an https redirect in future.
+    # - port: 80
+    #  name: http
+    - port: 443
+      name: https
+      targetPort: 80
+      protocol: TCP
+  selector:
+    k8s-app: "{{ .Values.service.name }}-load-balancer"
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ .Values.service.name }}-controller
+spec:
+  replicas: {{ .Values.replicaCount }}
+  revisionHistoryLimit: 3
+  template:
+    metadata:
+      labels:
+        k8s-app: "{{ .Values.service.name }}-load-balancer"
+    spec:
+      terminationGracePeriodSeconds: 60
+      containers:
+        - name: {{ .Values.service.name }}-controller
+          image: {{ .Values.image.repository }}:{{ .Values.image.version }}
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          readinessProbe:
+            httpGet:
+              path: /healthz
+              port: 18080
```

Comment:
This was also from the example, but from the nginx.conf the controller container generates:

```
# default server, used for NGINX healthcheck and access to nginx stats
    server {
        # Use the port 18080 (random value just to avoid known ports) as default port for nginx.
        # Changing this value requires a change in:
        # https://github.com/kubernetes/contrib/blob/master/ingress/controllers/nginx/nginx/command.go#L104
        listen 18080 default_server reuseport backlog=511;
        listen [::]:18080 default_server reuseport backlog=511;
```